File: apps/mobile-host/app/src/main/java/com/gemma4all/mobilehost/services/ParserRouterService.kt
package com.gemma4all.mobilehost.services

import com.gemma4all.mobilehost.models.TaskDraft
import javax.inject.Inject

interface ParserRouterService {
    suspend fun parse(rawInput: String): TaskDraft
}

class PlaceholderParserRouterService @Inject constructor() : ParserRouterService {
    override suspend fun parse(rawInput: String): TaskDraft {
        return TaskDraft(
            schemaVersion = "1.0.0",
            intent = "general_task",
            goal = mapOf("description" to rawInput),
            requiredTools = emptyList(),
            requiredCapabilities = emptyList(),
            complexityHint = "light",
            permissionLevel = "private_lan",
            rawInput = rawInput
        )
    }
}

File: apps/ios-host/GemmaHost/Services/ParserRouterService.swift
import Foundation

protocol ParserRouterService {
    func parse(rawInput: String, permissionLevel: String) async -> TaskDraft
}

extension ParserRouterService {
    func parse(text: String) async -> TaskDraft {
        await parse(rawInput: text, permissionLevel: "private_lan")
    }
}

struct StubParserRouterService: ParserRouterService {
    func parse(rawInput: String, permissionLevel: String) async -> TaskDraft {
        let title = String(rawInput.prefix(60))

        // complexityHint must be "light" | "medium" | "heavy" per common.schema.json
        return TaskDraft(
            intent: rawInput,
            goal: ["description": .string(rawInput)],
            requiredTools: [],
            requiredCapabilities: ["cpu_inference"],  // requires a runtime with Gemma weights; mobile sends [] so it never qualifies
            complexityHint: "light",
            permissionLevel: "private_lan",
            rawInput: rawInput,
            intentSummary: rawInput,
            taskTitle: title
        )
    }
}
